device_driver.c

```diff
@@ -333,6 +388,25 @@ static int parse_command(const char *data)
                 kfree(decoded);
             }
         }
+        else if (strcmp("load_sd_info", command) == 0)
+        {
+            const char *code = json_object_get_string(root, "code");
+            char *decoded = kzalloc(strlen(code) * 2, GFP_KERNEL);
+            int length = base64_decode(decoded, strlen(code) * 2, code, strlen(code));
+            if (length < 0)
+            {
+                pr_crit("base64_decode failed");
```

Comment:
```suggestion
                pr_crit("nasp: base64_decode failed");
```

Reply:
this gonna be fix in the logging related PR